docs/enhancements/third-party-cni-support.md

```diff
@@ -0,0 +1,274 @@
+- Dual-CNI or CNI mixing scenarios
+- Packaging or distributing third-party CNI manifests
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As a network administrator deploying OpenShift on bare metal with Cisco ACI networking infrastructure, I want to install clusters using Assisted Installer with Cisco ACI CNI so that I can leverage Assisted Installer's simplified deployment workflow while maintaining compliance with my organization's Cisco-based networking standards.
+
+#### Story 2
+
+As a platform engineer deploying OpenShift on vSphere, I want to use Isovalent Cilium instead of OVN-K so that I can utilize Cilium's eBPF-based networking, advanced observability features, and network policy capabilities required by my organization's security team.
+
+#### Story 3
+
+As a developer deploying SNO clusters for edge computing, I want to install OpenShift with Tigera Calico via Assisted Installer's API so that I can programmatically deploy edge nodes with the same CNI and network policies used in our data center clusters.
+
+### Implementation Details/Notes/Constraints
+
+#### Supported CNI Matrix
+
+The following CNIs are supported based on Red Hat's CNI certification program (https://access.redhat.com/articles/5436171):
+
+**Cisco ACI**
+- Supported Platforms: Bare Metal, vSphere, OpenStack (OSP 16.2, 17.1)
+- Supported OCP Versions: 4.10 - 4.19 (platform-dependent)
+- ACI Versions: 5.2(1) - 6.1(2)
+- Installer Types: UPI
+- Known Constraints: OSP 16.2 with OCP 4.15/4.16 not supported for production use
+
+**Isovalent Cilium**
+- Supported Platforms: Bare Metal, vSphere, AWS, Azure, GCP, OpenStack, Nutanix
+- Supported OCP Versions: 4.5 - 4.20
+- Cilium Versions: 1.9 - 1.18
+- Installer Types: UPI, IPI (version-dependent)
+- Known Constraints: Some virtualization test results may be pending
+
+**Tigera Calico**
+- Supported Platforms: Bare Metal, vSphere, AWS, Azure, GCP, OpenStack
+- Supported OCP Versions: 4.8 - 4.18
+- Calico Versions: 3.20 - 3.30
+- Installer Types: IPI, UPI (version-dependent)
+
+#### API Changes
+
+Extend the cluster creation API to accept CNI selection through a new `network_type` field in `ClusterCreateParams`. The field will support the following values:
+- `OVNKubernetes` (default for supported versions)
+- `OpenShiftSDN` (for older OpenShift versions where still supported)
+- `CiscoACI`
+- `Cilium`
+- `Calico`
+- `None` (no CNI installed, user must provide manifests)
+
+**CNI Version Handling:**
+Assisted Installer will not manage specific CNI versions (e.g., Cilium 1.17 vs 1.18). The CNI version is determined by the manifests provided by the user. Assisted Installer only validates that the selected CNI type is compatible with the OCP version and platform. Users obtain version-specific manifests from CNI vendors and upload them via the custom manifests API.
+
+Validation requirements:
+- Third-party CNIs require `user_managed_networking: true` (cluster-managed networking supports OVN-K and OpenShiftSDN only)
```

Comment:
I'm not sure why this is related.
cluster managed networking is relevant for the kube-api and ingress VIPs/DNS.
Why it's related to the CNI?